python_tsp/heuristics/local_search.py

```diff
@@ -78,12 +78,19 @@ def solve_tsp_local_search(
         improvement = False
         for n_index, xn in enumerate(neighborhood_gen[perturbation_scheme](x)):
             if default_timer() - tic > max_processing_time:
-                logger.warning("Stopping early due to time constraints")
+                warning_msg = "Stopping early due to time constraints"
+                if log_file:
+                    logger.warning(warning_msg)
+                print(f"WARNING: {warning_msg}")
```

Comment:
One more thing: instead of simply printing the message, can you add the `verbose` argument as you suggested initially and it set to `False` as default?

With that, the `print` would only happen if it was set to `True`:

```suggestion
                if verbose:
                    print(f"WARNING: {warning_msg}")
```